ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -137,3 +137,7 @@ static void *call_serialize_without_gvl(void *call_args) {
 
   return NULL; // Unused
 }
+
+void enforce_recorder_instance(VALUE object) {
+  Check_TypedStruct(object, &stack_recorder_typed_data);
```

Comment:
I couldn't find more information about `Check_TypedStruct`, and looking at the source didn't help much. Would you have any links around this?

Reply:
Yes!

These APIs are defined in https://github.com/ruby/ruby/blob/master/include/ruby/internal/core/rtypeddata.h . In particular `Check_TypedStruct` is an alias of `rb_check_typeddata` which you can find documented in https://github.com/ruby/ruby/blob/60d45b2ee86a80e248c3bff0c90c981ed2168ac3/include/ruby/internal/core/rtypeddata.h#L408-L429 .

Effectively [it's a variant of `Check_Type`](https://github.com/ruby/ruby/blob/master/doc/extension.rdoc) which is able to check for typedstructs (e.g. objects that wrap a native pointer) vs just built-in types as the regular `Check_Type` does.